source/containers.tex

```diff
@@ -9279,6 +9285,64 @@
 The specialization is enabled\iref{unord.hash}.
 \end{itemdescr}
 
+\indexlibraryglobal{\exposid{is-vector-bool-reference}}%
+\begin{itemdecl}
+template<class R>
+  inline constexpr bool @\exposid{is-vector-bool-reference}@ = @\seebelow@;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+The variable template
+\tcode{\exposid{is-vector-bool-reference}<T>} is \tcode{true}
+if \tcode{T} denotes the type \tcode{vector<bool, Alloc>::reference}
+for some type \tcode{Alloc} and
+\tcode{vector<bool, Alloc>} is not a program-defined specialization.
+\end{itemdescr}
+
+% FIXME: Where are these codeblocks supposed to be part of?
```

Comment:
Would like more direction on how to do this. This affects quite a few chunks of the paper. If it requires moving code around, it might need to wait until after LWG Motion 37 is applied since that paper depends on this one.

Reply:
Split the [vector.bool] subclause into [vector.bool.general] containing paras 1-3, and [vector.bool.ref] containing paras 4-6 and the _is-vector-bool-reference_ variable template. We need to find somewhere for `hash<vector<bool, Allocator>>`, maybe after para 3 in p[vector,bool.general]. Then the `formatter` specialization and its members can be in [vector.bool.formatter].

Reply:
I like your ideas, but sounds like it's is beyond the scope of this edit.

Reply:
Since this is only talking about introducing new subheadings (not changing any normative wording), I think it's fine to do those edits here.

Reply:
Ok to create a separate edit issue for this? Because:
* motions-2022-07-lwg-37 is based on this, so additional edits here would create conflicts there.
* There are other lonely code bits introduced by this paper that also need attention.
* We're already late for applying the motions, and I don't want to hold things up any longer.

Reply:
Opened #5719